libraries/SD/src/utility/SdFat.h

```diff
@@ -431,14 +431,16 @@ union cache_t {
 class SdVolume {
  public:
   /** Create an instance of SdVolume */
-  SdVolume(void) :allocSearchStart_(2), fatType_(0) {}
+  SdVolume(void) :allocSearchStart_(2), fatType_(0) { if (!cacheBuffer_) cacheBuffer_ = (cache_t *)malloc(sizeof(cache_t)); }

```

Comment:
null is not checked. I don't really know myself what to do in constructors when this happen.
Maybe you can also help with this in #4134 :)

Reply:
I believe errors in constructors can only throw an exception.  Unfortunately, that's not the Arduino way, so I'm as at a loss as yourself.

The alternative here, and elsewhere, is to use lazy allocation and/or defensive.  Before use of the cache it can be checked and the method can error out if it can't be allocated, assuming the API allows it.  Perusing the SD headers, it looks like SEGV'ing is really the only thing that's possible using the current API.

Something like link time optimization (-flto) might eliminate the need for this kind of change as it should be able to detect no call graph into this static class variable, but I've not been able to make it work (and I believe we're upgrading GCC soon, and at that point it should be "automatic" anyway).